packages/components/src/components/queryBuilder/index.tsx

```diff
@@ -0,0 +1,126 @@
+import { ChevronUpIcon } from '@heroicons/react/solid'
+import { atom, useAtom, WritableAtom } from 'jotai'
+import { useCallback } from 'preact/hooks'
+import { Resizable as Resize } from 're-resizable'
+import React from 'react'
+import { typesAtom } from '../provider'
+import { editorAtom, queryBuilderOpened } from '../tab/store'
+import { getDefaultOperation } from './getDefaultOperation'
+
+const Resizable = Resize as any
+
+const generatedAtom = atom<string | null>(null)
+const baseOperationNameAtom = atom<string | null>(null)
+// Updates `baseOperationNameAtom` and `generatedAtom`
+const operationNameAtom: WritableAtom<string | null, string> = atom(
+  get => get(baseOperationNameAtom),
+  (get, set, update) => {
+    set(baseOperationNameAtom, update)
+
+    const types = get(typesAtom)
+    if (!types) return
+
+    const defaultOp = getDefaultOperation({ types, operationName: update })
+    if (defaultOp?.value)
+      set(generatedAtom, defaultOp.value)
+  }
+)
+
+export const QueryBuilder: React.FC = () => {
+  const [types] = useAtom(typesAtom)
+  const [queryBuilderOpen, setQueryBuilderOpened] = useAtom(queryBuilderOpened)
+  const [generated] = useAtom(generatedAtom)
+  const [operationName, setOperationName] = useAtom(operationNameAtom)
+  const [editorView] = useAtom(editorAtom)
+
+  // On insert btn click
+  const insertGenerated = useCallback(
+    () => {
+      if (!operationName || !editorView) return
+      const gen = getDefaultOperation({ operationName, types })
+      if (!gen) return
+
+      const { value: generated, inputLength } = gen
+      const line = editorView.state.doc.lineAt(editorView.state.selection.main.head)
+
+      editorView.focus()
+      // Add generated query to next line in editor
+      editorView.dispatch({
+        changes: {
+          from: line.from,
+          to: line.to,
+          insert: `${line.text}\n${generated}`,
+        },
+      })
+
+      // Select generated default input so user can instantly remove/edit
```

Comment:
This doesn't seem to work, it's not selected for me

Reply:
That is weird, for me this is happening

![image](https://user-images.githubusercontent.com/67562518/182184272-4664df55-9ce2-4f01-a591-dcc09f256bc6.png)

Reply:
Hmm I'll test it again.